test/MockSmartStore.js

```diff
@@ -522,114 +522,128 @@ var MockSmartStore = (function(window) {
-                var paths = args[0].paths;
-                if (soupName == null) {errorCB("Bogus soup name: " + soupName); return;}
-                successCB(self.reIndexSoup(soupName, paths));
-            });
-
-            cordova.interceptExec(SMARTSTORE_SERVICE, "pgSoupExists", function (successCB, errorCB, args) {
-                var soupName = args[0].soupName;
-                successCB(self.soupExists(soupName));
-            });
-
-            cordova.interceptExec(SMARTSTORE_SERVICE, "pgQuerySoup", function (successCB, errorCB, args) {
-                var soupName = args[0].soupName;
-                var querySpec = args[0].querySpec;
-                successCB(self.querySoup(soupName, querySpec));
-            });
-
-            cordova.interceptExec(SMARTSTORE_SERVICE, "pgRunSmartQuery", function (successCB, errorCB, args) {
-                var querySpec = args[0].querySpec;
-                successCB(self.querySoup(null, querySpec));
-            });
-
-            cordova.interceptExec(SMARTSTORE_SERVICE, "pgRetrieveSoupEntries", function (successCB, errorCB, args) {
-                var soupName = args[0].soupName;
-                var entryIds = args[0].entryIds;
-                successCB(self.retrieveSoupEntries(soupName, entryIds));
-            });
-
-            cordova.interceptExec(SMARTSTORE_SERVICE, "pgUpsertSoupEntries", function (successCB, errorCB, args) {
-                var soupName = args[0].soupName;
-                var entries = args[0].entries;
-                var externalIdPath = args[0].externalIdPath;
-                successCB(self.upsertSoupEntries(soupName, entries, externalIdPath));
-            });
-
-            cordova.interceptExec(SMARTSTORE_SERVICE, "pgRemoveFromSoup", function (successCB, errorCB, args) {
-                var soupName = args[0].soupName;
-                var entryIds = args[0].entryIds;
-                self.removeFromSoup(soupName, entryIds);
-                successCB("OK");
-            });
-
-            cordova.interceptExec(SMARTSTORE_SERVICE, "pgMoveCursorToPageIndex", function (successCB, errorCB, args) {
-                var cursorId = args[0].cursorId;
-                var index = args[0].index;
-                successCB(self.moveCursorToPage(cursorId, index));
-            });
-
-            cordova.interceptExec(SMARTSTORE_SERVICE, "pgCloseCursor", function (successCB, errorCB, args) {
-                var cursorId = args[0].cursorId;
-                self.closeCursor(cursorId);
-                if (successCB) {
-                    successCB("OK");
-                }
-            });
-        }
-
     };
 
     // Return module
     return module;
 })(window);
 
+function hookToCordova(cordova, store, globalStore) {
+    var SMARTSTORE_SERVICE = "com.salesforce.smartstore";
+
+    cordova.interceptExec(SMARTSTORE_SERVICE, "pgGetDatabaseSize", function (successCB, errorCB, args) {
+        var targetStore = args[0].isGlobalStore ? store : globalStore;
```

Comment:
The call will be relayed to the regular MockSmartStore instance or to the global MockSmartStore instance.